Review for Project 3

chelmerrox opened the pull request:
I have completed this project.

My web page is divided as such:

1. The navbar that includes the login button

2. The main container that has:

   - Article 1
   - Article 2
   - Article 3
   - Article 4
   - Main container bottom links container (it's like a footer for the main container)
   - Read comments and bottom social media links container

Each article has its heading, images, captions etc. All of the 6 containers listed above act as grid items for the main (grid) container. 
So, the main container is divided into a 6 rows x 3 columns grid. It is a separate grid container.

3. The recommended articles container that has:

  - more space and astronomy container and editor's picks container
    a. more space and astronomy container is a 2 rows x 3 columns grid
    b. editor's picks container is a 1 row x 3 columns grid
  - more popular container (the sidebar)

The recommended articles container is divided into a 1 row x 2 columns grid.

4. Footer 1 container. It's divided as such:

  - Logo and go to home page container
  - footer 1 top links container (listed links in News, Opinion etc)
  -  footer 1 bottom links container (the web page ownership information in small texts)

5. Footer 2 container
  - footer 2 text
  - expand container

lekegitrepo commented:
#### Hi @chelmerrox :clap: :clap:
#### You have done an excellent job. There are few requirements missing in the project. Check my comment below :point_down: 

**_1. The "Theory of inflation" section is a combination of image and text check or inspects the original to understand the implementation._**

**_2. Elements should be aligned with a combination of all three techniques ([float](https://css-tricks.com/all-about-floats/), [flex](https://css-tricks.com/snippets/css/a-guide-to-flexbox/) and [grid](https://css-tricks.com/snippets/css/complete-guide-grid/)). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/html_css/positioning_and_floating_elements/flex_float_grid_requirement_pathwright.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59538403/)) and ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/html_css/positioning_and_floating_elements/flex_float_grid_requirement_odin.png) from [Odin](https://www.theodinproject.com/courses/html5-and-css3/lessons/positioning-and-floating-elements#introduction))_**
**_use of float is required, try to add this implementation in the CSS style rules_**

**_3. Use [`<aside>` tag](https://www.w3schools.com/tags/tag_aside.asp) for `Editor's picks` section on the righmost side. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/html_css/positioning_and_floating_elements/aside_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59538403/))_**

**_4. "More in Space and Astronomy" and "Most Popular" sections should contain links to other articles_**

**_Implement these required changes and submit them for review._**

**_if you have any doubts or questions contact me on slack._**

### _Reviewed by Temitayo Adeleke._

chelmerrox commented:
I have done the changes I needed to do for this project according to the previous TSEs' reviews:

1. Removed the image of the theory of inflation container and instead built it using grid. It is a 2 rows x 4 columns grid.

2. Used float and position properties in the expand container in footer 2 container.

3. Replaced div tag with the aside tag for the editors' picks container

4. Changed the tags used in all sub-containers in the recommended articles container to a tags and linking each article headings to their articles.

5. Removed comments in my index.html file.

rammazzoti2000 commented on index.html:
```diff
@@ -4,11 +4,1011 @@
     </head>
 
     <body>
+        <nav>
```
The social buttons are overlaying over the navbar when scrolling, this is due to the position property, which usually must be used with z-index property.
Please add `z-index: 2;`  to nav element

chelmerrox replied:
Thanks! I have resolved this problem.

rammazzoti2000 commented on index.html:
```diff
@@ -4,11 +4,1011 @@
+                <a href="https://www.nytimes.com/section/science/space" class="space-and-cosmos-link">SPACE &amp; COSMOS</a>
+            </div>
 
+            <img src="images/nyt_logo.PNG" alt="New York Times logo" width="170" height="30"/>
```
The NY Times logo is a [link to the page](https://www.nytimes.com/2014/03/18/science/space/detection-of-waves-in-space-buttresses-landmark-theory-of-big-bang.html?_r=0)

chelmerrox replied:
Thanks! I have resolved this problem.

rammazzoti2000 commented on index.html:
```diff
@@ -4,11 +4,1011 @@
+                        <p class="article-date">March 17, 2014</p>
+
+                        <!--The code for the icons is from the source page itself: https://www.nytimes.com/2014/03/18/science/space/detection-of-waves-in-space-buttresses-landmark-theory-of-big-bang.html?_r=0-->
+                        <div class="top-social-media-container">
```
The social buttons are overlaying over the navbar when scrolling, this is due to the position property, which usually must be used with z-index property.
Please add `z-index: 1;` to `top-social-media-container`

chelmerrox replied:
Thanks! I have resolved this problem.

rammazzoti2000 commented on index.html:
```diff
@@ -4,11 +4,1011 @@
+                    </div>
+                    <br>
+                    
+                    <div class="thank-you-container">
```
`thank you container` should be aligned along with the main body and not with the page, try to move this container inside the `paragraph-styling` container and find a way to align it properly.

chelmerrox replied:
Thanks! I have resolved this problem.

rammazzoti2000 commented on index.html:
```diff
@@ -4,11 +4,1011 @@
+
+                <div class="space-and-astronomy-container">
+                    
+                    <div class="space-and-astronomy-container-grid-item-1">
```
The description of each article in `More in Space and Astronomy` should be better aligned, the text is overflowing outside the container. Group the image and description together to limit the width of the description and not overflowing

rammazzoti2000 commented on index.html:
```diff
@@ -4,11 +4,1011 @@
+                        <p class="dates">April 17</p>
+                    </div>
+                    
+                    <div class="space-and-astronomy-container-grid-item-2">
```
The description of each article in `More in Space and Astronomy` should be better aligned, the text is overflowing outside the container. Group the image and description together to limit the width of the description and not overflowing

rammazzoti2000 commented on index.html:
```diff
@@ -4,11 +4,1011 @@
+                        <p class="dates">April 27</p>
+                    </div>
+                    
+                    <div class="space-and-astronomy-container-grid-item-3">
```
The description of each article in `More in Space and Astronomy` should be better aligned, the text is overflowing outside the container. Group the image and description together to limit the width of the description and not overflowing

rammazzoti2000 commented on index.html:
```diff
@@ -4,11 +4,1011 @@
+                        <p class="dates">April 17</p>
+                    </div>
+                    
+                    <div class="space-and-astronomy-container-grid-item-4">
```
The description of each article in `More in Space and Astronomy` should be better aligned, the text is overflowing outside the container. Group the image and description together to limit the width of the description and not overflowing

rammazzoti2000 commented on index.html:
```diff
@@ -4,11 +4,1011 @@
+                        <p class="dates">April 13</p>
+                    </div>
+                    
+                    <div class="space-and-astronomy-container-grid-item-5">
```
The description of each article in `More in Space and Astronomy` should be better aligned, the text is overflowing outside the container. Group the image and description together to limit the width of the description and not overflowing

chelmerrox replied:
Thanks! I have resolved this problem.

rammazzoti2000 commented on index.html:
```diff
@@ -4,11 +4,1011 @@
+                        <p class="dates">April 14</p>
+                    </div>
+                    
+                    <div class="space-and-astronomy-container-grid-item-6">
```
The description of each article in `More in Space and Astronomy` should be better aligned, the text is overflowing outside the container. Group the image and description together to limit the width of the description and not overflowing

rammazzoti2000 commented on index.html:
```diff
@@ -4,11 +4,1011 @@
+                    <h4 class="recommended-articles-headings">Editors' Picks</h4>
+
+                    <div class="editors-picks-container">
+                        <div class="editors-picks-container-grid-item-1">
```
The description of each article in `Editor's picks` should be better aligned, the text is overflowing outside the container. Group the image and description together to limit the width of the description and not overflowing

chelmerrox replied:
Thanks! I have resolved this problem.

rammazzoti2000 commented on index.html:
```diff
@@ -4,11 +4,1011 @@
+                            <p class="dates">April 28</p>
+                        </div>
+                        
+                        <div class="editors-picks-container-grid-item-2">
```
The description of each article in `Editor's picks` should be better aligned, the text is overflowing outside the container. Group the image and description together to limit the width of the description and not overflowing

rammazzoti2000 commented on index.html:
```diff
@@ -4,11 +4,1011 @@
+                            <p class="dates">April 22</p>
+                        </div>
+                        
+                        <div class="editors-picks-container-grid-item-3">
```
The description of each article in `Editor's picks` should be better aligned, the text is overflowing outside the container. Group the image and description together to limit the width of the description and not overflowing

chelmerrox replied:
Thanks! I have resolved this problem.

chelmerrox commented:
I have done the following changes according to the previous TSEs' review:

 1. Fixed the overlay of social media buttons on navbar when scrolling.

 2. Linked the NYT Logo to the home page.

 3. Aligned the thank-you-container along with the main body

 4. Fixed the alignment of the title description of each article in recommended articles container by 
   containing each grid item in a div with a class and adding width to it in the style.css file.

mikenath223 commented:
### Pull request currently under review by Michgolden Ukeje (TSE).
Kindly exercise patience 👋
